addons/TreasurePool/TreasurePool.lua

```diff
@@ -0,0 +1,171 @@
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL KENSHI BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.]]
+
+_addon.name = 'TreasurePool'
+_addon.author = 'Kenshi'
+_addon.version = '1.0'
+
+require('luau')
+texts = require('texts')
+packets = require('packets')
+
+-- Config
+
+defaults = {}
+defaults.display = {}
+defaults.display.pos = {}
+defaults.display.pos.x = 0
+defaults.display.pos.y = 0
+defaults.display.bg = {}
+defaults.display.bg.red = 0
+defaults.display.bg.green = 0
+defaults.display.bg.blue = 0
+defaults.display.bg.alpha = 102
+defaults.display.bg.visible = true
+defaults.display.text = {}
+defaults.display.text.font = 'Consolas'
+defaults.display.text.red = 255
+defaults.display.text.green = 255
+defaults.display.text.blue = 255
+defaults.display.text.alpha = 255
+defaults.display.text.size = 12
+
+settings = config.load(defaults)
+settings:save()
+
+treasure = texts.new(settings.display, settings)
+
+treasure:appendline('Treasure Pool:')
+for i = 0, 9 do
+    treasure:appendline(i .. ': ${index' .. i .. '|-}${lotting' .. i .. '}')
+end
+
+goals = {}
+
+lotter = {}
+
+lot = {}
+
+windower.register_event('incoming chunk', function(id, data)
+
+    if id == 0x0D2 then
+ 
+    local packet = packets.parse('incoming', data)
+    local time_check = packet.Timestamp + 300
+    local diff = os.difftime(time_check, os.time())
+        -- Ignore gil drop
+        if packet.Item == 0xFFFF then
+            return
+        end
+        if diff <= 300 then
+            goals[packet.Index] = packet.Timestamp + 300
+            lotter[packet.Index] = ' '
+        else
+            goals[packet.Index] = os.time() + 300
+            lotter[packet.Index] = ' '
+        end
+    
+    end
+    
+    if id == 0x0D3 then
+ 
+    local lotpacket = packets.parse('incoming', data)
+ 
+        -- Ignore drop to a player or floored
+        if lotpacket.Drop ~= 0 then
+            return
+        else    
+            lotter[lotpacket.Index] = lotpacket['Highest Lotter Name']
+            lot[lotpacket.Index] = lotpacket['Highest Lot']
+        end
+    
+    end
+    
+    -- Check to hide text box if zoning with treasure up
+    if id == 0xB then
+        zoning_bool = true
+    elseif id == 0xA and zoning_bool then
+        zoning_bool = nil
+    end
+    
+end)
+
+windower.register_event('prerender', function()
+    local get_item = windower.ffxi.get_items()
```

Comment:
Since you only need treasure from this you can do:
```lua
    local treasure = windower.ffxi.get_items().treasure
```

Then use that directly in the following code.